app/src/main/java/org/schabi/newpipe/local/holder/PlaylistItemHolder.java

```diff
@@ -35,10 +41,26 @@ public PlaylistItemHolder(final LocalItemBuilder infoItemBuilder, final ViewGrou
     public void updateFromItem(final LocalItem localItem,
                                final HistoryRecordManager historyRecordManager,
                                final DateTimeFormatter dateTimeFormatter) {
+        if (!BookmarkFragment.isMultiSelect) {
+            selectCheckBox.setChecked(false);
+            selectCheckBox.setVisibility(View.INVISIBLE);
+        } else {
+            selectCheckBox.setVisibility(View.VISIBLE);
+        }
+
         itemView.setOnClickListener(view -> {
             if (itemBuilder.getOnItemSelectedListener() != null) {
                 itemBuilder.getOnItemSelectedListener().selected(localItem);
             }
+            if (BookmarkFragment.isMultiSelect) {
+                if (selectCheckBox.isChecked()) {
+                    selectCheckBox.setChecked(false);
+                } else {
+                    selectCheckBox.setChecked(true);
+                }
+            } else {
+                selectCheckBox.setChecked(false);
+            }
```

Comment:
What? Why are you inverting the check state each time the item is updated? The item is updated at times you can't control, try to see for example what happens if you have a long list where items do not fit in a single screen but you have to scroll through them. Search for how to handle multiselection in RecyclerViews.